src/lib.rs

```diff
@@ -95,7 +95,7 @@ impl Stack {
                 ptr::null_mut(),
                 0x200000,
                 libc::PROT_READ | libc::PROT_WRITE,
-                libc::MAP_PRIVATE | libc::MAP_ANONYMOUS | libc::MAP_STACK,
+                libc::MAP_PRIVATE | libc::MAP_ANONYMOUS,
```

Comment:
From man page:
> This flag is currently a no-op on Linux.  However, by
              employing this flag, applications can ensure that they
              transparently obtain support if the flag is implemented in
              the future.  Thus, it is used in the glibc threading
              implementation to allow for the fact that some
              architectures may (later) require special treatment for
              stack allocations.  A further reason to employ this flag
              is portability: MAP_STACK exists (and has an effect) on
              some other systems (e.g., some of the BSDs).

So I think it's still best to keep it. Maybe use conditional compilation to gate it for OS X?